### ISSUE TITLE
Unable to start container

### ISSUE THREAD

**Issue by unkcpz** [aiidalab/aiidalab-docker-stack#223]
I was able to start the container with `./manage.py --develop up`. But after I reboot my computer, I got the following message every time I try to start the container:

```
(aiidalab-dev) ╭─jyu@theospc38 in ~/Projects/Marketplace/aiidalab-docker-stack on develop ✔ (origin/develop)
╰$ ./manage.py --develop up                                    
Starting AiiDAlab (this can take multiple minutes) ...
Unable to communicate with the AiiDAlab container. Is it running? Use `up` to start it.
```

My `.env` is:
```
╰$ cat .env
AIIDALAB_HOME_VOLUME=~/aiidalab
AIIDALAB_PORT=8888
AIIDALAB_DEFAULT_APPS=quantum-espresso==21.10.0b1
JUPYTER_TOKEN=ea37514ecee08e1e71bf81e5f84e16cf2ab9dd286e23cc3652412174f04c3e70
SYSTEM_USER=aiida
```

@csadorf any advice on how to find the problem?

**Comment by csadorf** [comment]
@unkcpz Can you run with `-v` / `--verbose` set to inspect the log output and also post the output of `docker-compose logs -f aiidalab`?

**Comment by unkcpz** [comment]
@csadorf nothing special with `-v`. But the `docker-compose logs -f aiidalab` shows:

```
...
aiidalab_1  | ----------------------  -----------------------------------------------------------------------------------------------------------
aiidalab_1  | + verdi daemon stop
aiidalab_1  | Profile: default
aiidalab_1  | Warning: Deleted apparently stale daemon PID file as its associated process<1277> does not exist anymore
aiidalab_1  | Waiting for the daemon to shut down... The daemon was not running.
aiidalab_1  | + verdi database migrate --force
aiidalab_1  | + verdi daemon start
aiidalab_1  | Starting the daemon... RUNNING
aiidalab_1  | *** Running /etc/my_init.d/80_prepare-aiidalab.sh...
aiidalab_1  | + export SHELL=/bin/bash
aiidalab_1  | + SHELL=/bin/bash
aiidalab_1  | + '[' '!' -e /home/aiida/.ipython/profile_default/startup/aiida_magic_register.py ']'
aiidalab_1  | + '[' '!' -e /home/aiida/apps ']'
aiidalab_1  | + '[' '!' -e /home/aiida/apps/home ']'
aiidalab_1  | + [[ -d /home/aiida/apps/home ]]
aiidalab_1  | + [[ ! -L /home/aiida/apps/home ]]
aiidalab_1  | + [[ '' == 1 ]]
aiidalab_1  | + reentry scan
aiidalab_1  | + '[' -e /home/aiida/.trash ']'
aiidalab_1  | + find /home/aiida/.trash/ -mindepth 1 -delete
aiidalab_1  | find: cannot delete ‘/home/aiida/.trash/aiidalab-sssp-6212ee9e-ed3d-40b9-995d-e8cbd266ec6d/.git/objects/c5/b3d1ecc51441ef148c8d13965cae568e2b4249’: Permission denied
aiidalab_1  | find: cannot delete ‘/home/aiida/.trash/aiidalab-sssp-6212ee9e-ed3d-40b9-995d-e8cbd266ec6d/.git/objects/c5’: Directory not empty
aiidalab_1  | find: cannot delete ‘/home/aiida/.trash/aiidalab-sssp-6212ee9e-ed3d-40b9-995d-e8cbd266ec6d/.git/objects/a0/5de25dfa1e4d3fd201790625b114fd10984e62’: Permission denied
aiidalab_1  | find: cannot delete ‘/home/aiida/.trash/aiidalab-sssp-6212ee9e-ed3d-40b9-995d-e8cbd266ec6d/.git/objects/a0’: Directory not empty
aiidalab_1  | find: cannot delete ‘/home/aiida/.trash/aiidalab-sssp-6212ee9e-ed3d-40b9-995d-e8cbd266ec6d/.git/objects/30/1f35b988e887f13368ae8778d414e3317cea73’: Permission denied
aiidalab_1  | find: cannot delete ‘/home/aiida/.trash/aiidalab-sssp-6212ee9e-ed3d-40b9-995d-e8cbd266ec6d/.git/objects/30’: Directory not empty
...
...
aiidalab_1  | *** /etc/my_init.d/80_prepare-aiidalab.sh failed with status 1
aiidalab_1  | 
aiidalab_1  | *** Killing all processes...
aiidalab_1  | Nov 15 12:41:50 861eb19bb2c9 syslog-ng[116]: syslog-ng shutting down; version='3.13.2'
```

**Comment by unkcpz** [comment]
I think I can try to clean the `.trash` directory before starting. But what is the purpose of the line `find /home/aiida/.trash/ -mindepth 1 -delete`?

**Comment by csadorf** [comment]
> I think I can try to clean the `.trash` directory before starting. But what is the purpose of the line `find /home/aiida/.trash/ -mindepth 1 -delete`?

The script runs a few maintenance tasks (including the emptying of the trash directory) that should not be executed concurrently. We can assume that there is only one init process so that's a safe place to do that. That said, we probably have to make that command more robust against permission errors. I am currently checking the `find` man page.

**Comment by unkcpz** [comment]
After I clean the `.trash/` and rerun the `manage.py` I got:

```
...
aiidalab_1  | + verdi daemon stop
aiidalab_1  | Profile: default
aiidalab_1  | Warning: Deleted apparently stale daemon PID file as its associated process<1276> does not exist anymore
aiidalab_1  | Waiting for the daemon to shut down... The daemon was not running.
aiidalab_1  | + verdi database migrate --force
aiidalab_1  | + verdi daemon start
aiidalab_1  | Starting the daemon... RUNNING
aiidalab_1  | *** Running /etc/my_init.d/80_prepare-aiidalab.sh...
aiidalab_1  | + export SHELL=/bin/bash
aiidalab_1  | + SHELL=/bin/bash
aiidalab_1  | + '[' '!' -e /home/aiida/.ipython/profile_default/startup/aiida_magic_register.py ']'
aiidalab_1  | + '[' '!' -e /home/aiida/apps ']'
aiidalab_1  | + '[' '!' -e /home/aiida/apps/home ']'
aiidalab_1  | + [[ -d /home/aiida/apps/home ]]
aiidalab_1  | + [[ ! -L /home/aiida/apps/home ]]
aiidalab_1  | + [[ '' == 1 ]]
aiidalab_1  | + reentry scan
aiidalab_1  | + '[' -e /home/aiida/.trash ']'
aiidalab_1  | + find -L /home/aiida -maxdepth 3 -name apps_meta.sqlite -delete
aiidalab_1  | find: File system loop detected; ‘/home/aiida/aiida’ is part of the same file system loop as ‘/home/aiida’.
aiidalab_1  | *** /etc/my_init.d/80_prepare-aiidalab.sh failed with status 1
aiidalab_1  | 
aiidalab_1  | *** Killing all processes...
aiidalab_1  | Nov 15 12:48:07 53e6cb9c8379 syslog-ng[115]: syslog-ng shutting down; version='3.13.2'
aiidalab-docker-stack_aiidalab_1 exited with code 1
```

**Comment by csadorf** [comment]
> After I clean the `.trash/` and rerun the `manage.py` I got:
> 
[...]

This bug was originally also reported on slack by @mbercx . I've created a separate issue for it: #225

**Comment by csadorf** [comment]
@unkcpz I've created #224 to resolve the originally reported problem, would you mind testing it?

**Comment by unkcpz** [comment]
I am sorry that I remove the `.trash/` folder and was unable to reproduce the issue. And I can not start the aiidalab now. I Will test it once I encounter it again
.

**Comment by csadorf** [comment]
> I am sorry that I remove the `.trash/` folder and was unable to reproduce the issue. And I can not start the aiidalab now. I Will test it once I encounter it again. If I remember it correctly, this `.trash` folder is create when I click `uninstall` for app in app store.

Yes, no problem. You might be able to reproduce the issue by either simply creating a directory and file within the `.trash/` directory with root permissions or by running `aiidalab uninstall` with root permissions. I assume you ran `aiidalab uninstall` from within the container, e.g., within a vscode terminal or so?
